background/storage/local.ts

```diff
@@ -69,11 +69,29 @@ export async function getStorage() {
       { override: true }
     )
     storage.subscribe(async (state) => {
-      console.log("[background] Sync state to popup")
-      await browser.runtime.sendMessage(browser.runtime.id, {
-        action: StorageAction.Sync,
-        state
-      })
+      console.log(
+        "[background] Sync state to app/storage/index.ts (in tabs/index.html)"
+      )
+      // Avoid "Receiving end does not exist" error due to missing app-side addListener.
+      try {
+        await browser.runtime.sendMessage(browser.runtime.id, {
+          action: StorageAction.Sync,
+          state
+        })
+      } catch (e) {
+        if (
+          e instanceof Error &&
+          e.message.includes(
+            `Could not establish connection. Receiving end does not exist.`
+          )
+        ) {
+          console.warn(`An error occurred while receiving end: ${e}`)
+        } else {
+          console.error(
+            `An unexpected error occurred while receiving end: ${e}`
+          )
+        }
```

Comment:
The condition seems identical to me. You can directly log the error with `warn` or `error`

Reply:
Got it!
I will replace the entire `if`-`else` condition with ```console.warn(`An error occurred while receiving end: ${e}`)``` to differentiate between `expected` and `unexpected` errors.